src/test/java/unit/BqDatasetNumTables.java

```diff
@@ -153,5 +176,167 @@ void numTablesForReferencedWithNoAccess() throws IOException {
     // the external dataset created in the beforeall method should have 1 table in it, but the
     // sharee user doesn't have read access to the dataset so they can't know that
     assertNull(describeDataset.numTables, "referenced dataset with no access contains NULL tables");
+
+    // Clean up.
+    workspaceCreator.login();
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + createWorkspace.id);
+    TestCommand.runCommandExpectSuccess("resource", "delete", "--name=" + name, "--quiet");
+    TestCommand.runCommandExpectSuccess("workspace", "delete", "--quiet");
+  }
+
+  @Test
+  @DisplayName("Attempt to update table reference but the user has no access.")
+  void updateTableReferenceWithNoAccess() throws IOException {
```

Comment:
I don't think these tests for updating the reference belong in this test class `BqDatasetNumTables`. Suggest moving them to a new class e.g. `BqDatasetReferencedUpdate`.